### PR TITLE
[SDK]  Fix flow as function connection override when node has default variant.

### PR DESCRIPTION
**Pull request by D-W-**
# Description

This pull request primarily concerns the handling of flow context configurations and their potential interactions in the PromptFlow package. The changes include additional documentation on potential issues with using multiple overrides, a bug fix for flow function connection override when a node has a default variant, and adjustments to the flow context resolver and test cases to accommodate these changes.

Documentation Updates:

* [`docs/how-to-guides/execute-flow-as-a-function.md`](diffhunk://#diff-b01a987e6744c2de5b19ecd68893c154271b0350749ad6512dc7de20981e7a4fR87-R99): Added a section on flow with multiple overrides, warning users about potential undefined behavior when using `connection` and `overrides` to override the same node.
* [`examples/tutorials/get-started/flow-as-function.ipynb`](diffhunk://#diff-6344b82d7eea09b34e46391271f05c9efaa24c09c16d6bd2867196bb32d9a55cL19-R23): Added a note about the potential issues with flow context configurations affecting each other.

Bug Fixes:

* [`src/promptflow/CHANGELOG.md`](diffhunk://#diff-41ec3f7c4b5d4c0e670407d3c00a03a6966d7ebf617b1536473e33a12e2bc765R17): Documented a bug fix for flow as function connection override when a node has a default variant.

Code Adjustments:

* [`src/promptflow/promptflow/_sdk/operations/_flow_context_resolver.py`](diffhunk://#diff-fa095e2bc6aacfd65f8ce4272436ad57636b92a08c143a8b411393ee6a52320bR55-L61): Modified the `_resolve_variant` method to import `overwrite_variant` only when necessary and adjusted the handling of the `flow_context.variant` attribute.
* [`src/promptflow/tests/sdk_cli_test/e2etests/test_flow_as_func.py`](diffhunk://#diff-164e59c72526e78234995ba0af35845687c275bbd556145539746197d6067ef7R242-R251): Added a test case for a flow with a default variant, testing that the function can run successfully with a connection override.
# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by wangchao1230 — APPROVED** [review]

**Review by YingChen1996 — DISMISSED** [review]

File: src/promptflow/promptflow/_sdk/operations/_flow_context_resolver.py [hunk 1]
```diff
@@ -52,13 +52,14 @@ def _resolve(self, flow_context: FlowContext):
     def _resolve_variant(self, flow_context: FlowContext) -> "FlowContextResolver":
         """Resolve variant of the flow and store in-memory."""
         # TODO: put all varint string parser here
+        from promptflow._sdk._submitter import overwrite_variant
+
         if not flow_context.variant:
             return self
+            tuning_node, variant = None, None
```
Comment (0mza987) [inline src/promptflow/promptflow/_sdk/operations/_flow_context_resolver.py:59]: will this be executed?

Comment (D-W-) [reply src/promptflow/promptflow/_sdk/operations/_flow_context_resolver.py:59]: Good catch!

**Review by 16oeahr — APPROVED** [review]

File: src/promptflow/tests/sdk_cli_test/e2etests/test_flow_as_func.py [hunk 2]
```diff
@@ -239,3 +239,14 @@ def test_flow_as_func_perf_test(self):
         f = load_flow(f"{FLOWS_DIR}/print_env_var")
         for i in range(100):
             f(key="key")
+
+    def test_flow_with_default_variant(self, azure_open_ai_connection):
+        f = load_flow(f"{FLOWS_DIR}/web_classification_default_variant_no_llm_type")
+        # f = load_flow(r"D:\code\prompt-flow\examples\flows\standard\web-classification\flow.dag.yaml")
```
Comment (wangchao1230) [inline src/promptflow/tests/sdk_cli_test/e2etests/test_flow_as_func.py:245]: remove this line?

**Review by wangchao1230 — APPROVED** [review]

**Review by brynn-code — APPROVED** [review]

File: scripts/installer/windows/product.wxs [hunk 3]
```diff
@@ -59,9 +59,9 @@
         <Icon Id="PromptflowIcon" SourceFile="$(var.ProductResources)logo32.ico" />
 
         <Property Id="ARPPRODUCTICON" Value="PromptflowIcon" />
-        <Property Id="ARPHELPLINK" Value="https://microsoft.github.io/promptflow/how-to-guides/quick-start.html" />
-        <Property Id="ARPURLINFOABOUT" Value="https://microsoft.github.io/promptflow/how-to-guides/quick-start.html" />
-        <Property Id="ARPURLUPDATEINFO" Value="https://microsoft.github.io/promptflow/how-to-guides/quick-start.html" />
+        <Property Id="ARPHELPLINK" Value="https://microsoft.github.io/promptflow/how-to-guides/quick-start/index.html" />
```
Comment (wangchao1230) [inline scripts/installer/windows/product.wxs:62]: why change this?

Comment (D-W-) [reply scripts/installer/windows/product.wxs]: Good catch, reverted.

**Review by brynn-code — APPROVED** [review]

**Review by crazygao — APPROVED** [review]

**Review by 0mza987 — APPROVED** [review]

**Pull request merged**